src/illuminatio/rule.py

```diff
@@ -18,17 +21,29 @@
 
 
 def dict_mapper(obj):
+    """
+    converts an object to a dictionary if possible
+    """
     return obj.to_dict() if hasattr(obj, "to_dict") else obj
 
 
 class Connection(Conn):
+    """
+    class for interaction with Connection tuples
+    """
     def to_dict(self) -> dict:
+        """
+        converts a Connection tuple into a dictionary
+        """
         return {self.direction: {"hosts": dict_mapper(self.targets),
                                  "ports": list(map(dict_mapper, self.ports))}}
 
 
 # main class
 class Rule:
+    """
+    class for networking rules
```

Comment:
Start with a capital letter to match the rest of the docstrings (see: https://www.python.org/dev/peps/pep-0257/)